transport/grpc/dial.go

```diff
@@ -138,7 +138,7 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 		// * The endpoint is a host:port (or dns:///host:port).
 		// * Credentials are obtained via GCE metadata server, using the default
 		//   service account.
-		if o.EnableDirectPath && checkDirectPathEndPoint(endpoint) && isTokenSourceDirectPathCompatible(creds.TokenSource) && metadata.OnGCE() {
+		if o.EnableDirectPath && checkDirectPathEndPoint(endpoint) && isTokenSourceDirectPathCompatible(creds.TokenSource, o) && metadata.OnGCE() {
```

Comment:
```
&& (o.AllowNonDefaultServiceAccount || isTokenSourceDirectPathCompatible(...)
``

Reply:
Sorry it seems this comment will change the functionality. For example if `creds.TokenSource` is `nil`, then `isTokenSourceDirectPathCompatible(creds.TokenSource, o)` will be `false`, but `(o.AllowNonDefaultServiceAccount || isTokenSourceDirectPathCompatible(creds.TokenSource))` could be `true` if `o.AllowNonDefaultServiceAccount` is true. 

Based on my understanding, oauth token is per-RPC and attached to the RPC header, while service account is per-connection and attached in the ALTS context when DirectPath is doing ALTS handshake. So we should not change the functionality here.